plugin_interpreter/src/controller_plugin.py

```diff
@@ -125,6 +126,51 @@ def start(self, logger, signal):
         """
         pass
 
+    def _advertise_functionality(self):
+        """Advertises functionality to database
+
+        This will send the contents of the self.functionality
+        attribute to the database interface. The table for
+        the plugin will be named the exact same string as the
+        self.name attribute.
+        """
+        self.db_send.put((self.name, self.functionality))
+
+    def _request_job(self):
+        """Request next job
+
+        This first checks the receive queue to see if there is
+        a job waiting, then if the queue is empty, it sends a
+        request to the database handler to reply with the next
+        new job whose start time is in the past.
+        
+        Returns:
+            {dict} -- a dictionary describing the job containing
+            {
+                "id": {string} -- GUID, not needed for plugin,
+                "JobTarget": {dict} -- target from Targets table,
+                "Status": {string} -- the status of the job,
+                "StartTime": {int} -- unix epoch start time,
+                "JobCommand": {dict} -- command to run
+            }
+        """
+        job = None
+        try:
+            job = self.db_recv.get_nowait()
+            return job
+        except Empty:
+            pass
+
+        self.db_send.put(self.name)
```

Comment:
consider putting this in a length 1 tuple so the receiver can check length to infer the question?